File: Python3/401 - 600/405 Convert a Number to Hexadecimal.py
# 36ms, 98.77%
class Solution:
    def toHex(self, num):
        if num == 0: return '0'
        num &= 0xFFFFFFFF
        hexalpha = ['0', '1', '2', '3', '4', '5', '6', '7',
                    '8', '9', 'a', 'b', 'c', 'd', 'e', 'f']
        res = []
        while num:
            tmp = num % 16
            num //= 16
            res.append(hexalpha[tmp])
        return ''.join(res[::-1])
    

File: 401 - 600/405 Convert a Number to Hexadecimal.cc
// 4ms, 82.80%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string toHex(int num) {
        if (num == 0) return "0";
        unsigned un_num = num;
        char hexalpha[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
        string res = "";
        while (un_num) {
            int tmp = un_num % 16;
            un_num /= 16;
            res += hexalpha[tmp];
        }
        return string(res.rbegin(), res.rend());
    }
};
